sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java

```diff
@@ -500,11 +502,12 @@
      *
      * @param options Optional. Options to filter configuration setting results from the service.
      * @param context Additional context that is passed through the Http pipeline during the service call.
-     * @return A List of ConfigurationSettings that matches the {@code options}. If no options were provided, the List
+     * @return A {@link Stream} of ConfigurationSettings that matches the {@code options}. If no options were provided, the List
      * contains all of the current settings in the service.
      */
-    private Iterable<ConfigurationSetting> listSettings(SettingSelector options, Context context) {
-        return client.listSettings(options, context).collectList().block();
+    private Stream<ConfigurationSetting> listSettings(SettingSelector options, Context context) {
+        List<ConfigurationSetting> listConfigurationSettings = client.listSettings(options, context).collectList().block();
+        return listConfigurationSettings != null ? listConfigurationSettings.stream() : Stream.empty();
```

Comment:
This is unnecessary extra work, just do `return client.listSettings(options, context).toStream()`